### ISSUE TITLE
GNOME Files: can't show audio/video informations

### ISSUE THREAD

**Issue by davidak** [NixOS/nixpkgs#53631]
## Issue description

I was able to reproduce it with MP3, WAV, WEBM, MP4. It probably happens with every file format.

![screenshot from 2019-01-08 08-26-46](https://user-images.githubusercontent.com/91113/50816072-b8643680-131f-11e9-847e-73bdb5abb649.png)

Maybe an optional dependency like ffmpeg is missing?

### Steps to reproduce

1. open GNOME Files
2. right click on audio or video file
3. click "Properties"
4. click on "Audio/Video" tab
## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.90, NixOS, 18.09.1799.b9fa31cea0e (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(davidak): `""`
 - channels(root): `"nixos-18.09.1799.b9fa31cea0e, nixos-hardware, nixos-unstable-19.03pre162671.44b02b52ea6"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by hedning** [comment]
I'm able to reproduce this on nixos-unstable.

**Comment by davidak** [comment]
>add the plugins to `nautilus`’s `buildInputs`

this sounds like the best solution. easy and reliable

**Comment by jtojnar** [comment]
GNOME 3.32 (#57027) moved the panel from Totem to Nautilus so now we are adding the plugins to `buildInputs`. Unfortunately, this bloats Nautilus even for people who do not need this.

Also we are only adding base for now. We will probably need to add more codecs for this to be useful.

**Comment by alexeymuranov** [comment]
Still an issue

**Comment by dasj19** [comment]
Still an issue in the upcoming Gnome 40: https://github.com/NixOS/nixpkgs/pull/117081

**Comment by bjornfor** [comment]
Does https://github.com/NixOS/nixpkgs/issues/195936#issuecomment-1278954466 have good merge semantics? Why don't we use that until someone has time for a cleaner solution?

**Comment by bjornfor** [comment]
> Does [#195936 (comment)](https://github.com/NixOS/nixpkgs/issues/195936#issuecomment-1278954466) have good merge semantics? Why don't we use that until someone has time for a cleaner solution?

No, multiple `environment.sessionVariables.GST_PLUGIN_SYSTEM_PATH_1_0 = "string"` definitions do not merge (eval error).

**Comment by bjornfor** [comment]
However, we can use lists instead. From `man configuration.nix` and search for `sessionVariables`: `The value of each session variable can be either a string or a list of strings. The latter is concatenated, interspersed with colon characters.`. Perfect!

**Comment by jtojnar** [comment]
The issue with that is that now you can have multiple incompatible GStreamer versions loaded at the same time. GStreamer claims to be [ABI compatible](https://gstreamer.freedesktop.org/documentation/frequently-asked-questions/developing.html?gi-language=c#what-kind-of-versioning-scheme-does-gstreamer-use) but [accidental breakage can occur](https://abi-laboratory.pro/index.php?view=timeline&l=gstreamer). So I would rather prefer avoiding setting binary modules to load globally.

**Comment by bjornfor** [comment]
@jtojnar: It would only break for people mixing stable + unstable channels, right?

**Comment by jtojnar** [comment]
Or if they use `nixos-rebuild switch` without restarting their system.

**Comment by jonasstaehle** [comment]
This affects me after upgrading from 20.04 LTS to 22.04 LTS
Is there anything i can do to fix the error?

**Comment by codethief** [comment]
Not stale

**Comment by l0b0** [comment]
> Since [#195936](https://github.com/NixOS/nixpkgs/issues/195936) is a duplicate of this, I'm cross posting the workaround shared there for visibility (and to link this tracking issue + workaround in my nixos config lol)
> 
> # in configuration.nix
> environment.sessionVariables.GST_PLUGIN_SYSTEM_PATH_1_0 = lib.makeSearchPathOutput "lib" "lib/gstreamer-1.0" [
>   pkgs.gst_all_1.gst-plugins-good
>   pkgs.gst_all_1.gst-plugins-bad
>   pkgs.gst_all_1.gst-plugins-ugly
>   pkgs.gst_all_1.gst-libav
> ]

Is it possible that this is breaking stuff in current unstable? Because my journal is flooded with lines like

> 'g_io_module_load': /nix/store/250qxwnb9ky20dgaxlmlp6s2il5xycbk-gst-plugins-good-1.26.11/lib/gstreamer-1.0/libgsttaglib.so: undefined symbol: g_io_module_load

Reverting this workaround seems to fix the issue.
